config/routes.rb

```diff
@@ -2,5 +2,13 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  root "users#index"
+
+  get "/users", to: "users#index"
+  get "/users/:id", to: "users#show"
+  get "/users/:id/posts", to: "posts#index"
+  get "/users/:id/posts/3", to: "posts#show"
+  # resources :users do
+  #   resources :posts
+  # end
```

Comment:
- [ ] I would recommend maintaining a RESTful URL and also the correct URL.

`get "/users/:id/posts/3", to: "posts#show"` :- this is kind of specific URL which only go to the post-show page whose post_id is 3. You can use what I have mentioned below.

```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
Thank you for this awesome tip @Dipeshtwis 🙏 . I was wondering how to do the nesting here without using resources but this is so much better. I have gone ahead and implemented that!